### ISSUE TITLE
Slow write, thinking mode

### ISSUE THREAD

**Issue by lud** [flide/8VIM#433]
**Is your feature request related to a problem? Please describe.**
When I am not sure what I need to type and need time to think, I do not like having to keep my finger in the circle. And more generally, the way I write has me untouching the screen very often, which insert space because the app interprets that as a touch on the circle.

**Describe the solution you'd like**
I would like to be able to enable a mode in the settings where:
* If I end a letter in the circle and raise my thumb back, nos space is inserted.
* If otherwise I continue for another letter the writing goes on as usual.
* With this mode, inserting a space when the thumb is up requires to tap on the circle. When writing multiple letters in a row, inserting a space requires to raise the thumb back and the tap the circle.
* Also this could enable inserting a letter by raising the finger when entered the final sector, without having to go back to the circle.

**Describe alternatives you've considered**
* Maybe a mode where space would never be inserted automatically at all, and would be inserted by crossing the circle from left to right (just like righ-to-left opens a selection/clipboard/move menu)

**Additional context**
Sorry If that has been discussed before, I did not found it in the issues. Also I tried to search if it was already possible in the settings but it seems that it does not.

I think this feature could be a good way to enable learning the app without inserting spaces all the time, and also solves the problem of the thumb/finger hiding the characters on the bottom right corner (for right-handed people) which makes the app harder to learn.

**Comment by fgudin** [comment]
I’ve been annoyed also for a while for the very same reason. However, there’s a quick (i mean, no code) work-around: just leave the circle and raise your thumb. If it matters, in my case, i just go south and raise my finger, voilà. Stops there, no space inserted. I don’t remember having customized anything in this regard.
Thanks, btw: i never took the time to learn that a mere circle tap would insert a space. So handy :)

**Comment by lud** [comment]
Hello @fgudin thanks for the tip, I'll try to use the keyboard again with your tip :)

**Comment by MaethorNaur** [comment]
It's something I'm aware of. I'm thinking about a smart space, depending on the language.
For instance, in French we add a space before semi-colon, but not in English.
Maybe automatically adding a space after a punctuation

**Comment by EricDru** [comment]
I suspect the old 8pen didn't auto-insert spaces, so maybe a setting for 8pen mode?

The YouTube video linked from 8vim help is a dead link. Searching found me the 8pen tutorial. Took a while to figure how not to s p a c e each letter.

Just a newb here. The option to hide letters suggests it gets better.

**Comment by ManDay** [comment]
> It's something I'm aware of. I'm thinking about a smart space, depending on the language. For instance, in French we add a space before semi-colon, but not in English. Maybe automatically adding a space after a punctuation

They say the way to hell is paved with "smart" software!

I think an option to disable space-on-center-finger-up is a fair compromise to onboard beginners, though I don't think it's particularly viable for real (power) usage. The system with the implied space is very efficient as it is.

**Comment by lud** [comment]
> The system with the implied space is very efficient as it is.

Yes but it also counts as "smart" software, and is really frustrating when you learn the app. Real power usage options should be opt-in once you get accustomed to the software in my opinion. But an opt-out version would already be very satisfying.

**Comment by MaethorNaur** [comment]
@EricDru I'm currently on #530 which should list all available characters/actions of a layout.
Also, you will be able to see which movement you need to do for that character/action.

It's a start, far from perfect but that might help newcomers for now.
We could develop that feature in the future.

@lud From the latest release (well version >=0.17) there is a new button called `fn`. More details [here](https://8vim.github.io/docs/guides/function-button).
The space movement is not included yet but can be easily be added
